File: 69.sqrt-x.py
#
# @lc app=leetcode id=69 lang=python3
#
# [69] Sqrt(x)
#

# @lc code=start
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:
            return 0

        l = 1
        r = x
        mid = 0
        while l <= r:
            mid = (l + r) // 2

            if x // mid == mid:
                return mid
            elif x // mid > mid:
                l = mid + 1
            else:
                r = mid - 1
        
        return r;
        
# @lc code=end

File: 69.sqrt-x.java
/*
 * @lc app=leetcode id=69 lang=java
 *
 * [69] Sqrt(x)
 */

// @lc code=start
class Solution {
    public int mySqrt(int x) {
        if (x == 0) {
            return 0;
        }
        int l = 1, r = x;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (mid == x / mid) {
                return mid;
            } else if (mid > x / mid) {
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        return r;
    }
    
}
// @lc code=end

